src/function_classes/chebychev_specials.py

```diff
@@ -0,0 +1,49 @@
+import torch
+import torch.distributions as D
+import numpy as np
+import random
+
+from core import FunctionClass
+
+class ChebychevSharedRoots(FunctionClass):
+
+    """
+    This class generates polynomials with shared roots
+    Roots can be perturbed
+    And the polynomials are mainly scaled so that the maximum value is 1
+    Although scaling_perc is the percentage of scaling that is uniform random
+    """
+
+    def __init__(self, degree, perturbation=0.1, *args, **kwargs):
+
+        self.perturbation = perturbation
+        self._one_minus_one = torch.tensor([-1, 1])
+
+        k = torch.arange(1, degree + 1)
+        self.chebychev_roots = torch.cos((2 * k - 1) * torch.pi / (2 * degree)).view(1, -1)
+        self.chebychev_roots = self.chebychev_roots.expand(kwargs['x_distribution'].batch_shape[0], -1)
+
+        super(ChebychevSharedRoots, self).__init__(*args, **kwargs)
+
+    def _init_param_dist(self) -> D.Distribution:
+        """Produce the distribution with which to sample parameters"""
+        perturbationd_dist = D.Uniform(-self.perturbation*torch.ones_like(self.chebychev_roots), self.perturbation*torch.ones_like(self.chebychev_roots))
+        return perturbationd_dist
+    
+    def evaluate(self, x_batch: torch.Tensor, params: torch.Tensor) -> torch.Tensor:
+
```

Comment:
match the other functions, pass in `*params`, then in the beginning of the function, add `weights, *_ = params`

Reply:
Fixed this in new coomit